### ISSUE TITLE
Actions object disappeared from embedded instance

### ISSUE THREAD

**Issue by rogerhyam** [ProjectMirador/mirador#3200]
I’ve been using Mirador 3 npm installs embedded into my react-bootstrap project for the last few months but it has just stopped working.

I basically do what it says on the GitHub readme

this.miradorInstance = mirador.viewer(this.initConfig, []);
const { store, actions } = this.miradorInstance;
store.dispatch(actions.fetchManifest(id));
store.dispatch(actions.removeManifest(id));

But after the latest npm update (I’m now on mirador@3.0.0-rc.3) the store instance no longer contains an actions object!

<img width="673" alt="Screenshot 2020-07-07 at 12 41 19" src="https://user-images.githubusercontent.com/1799028/86775865-a320b300-c04f-11ea-93a7-cd086a190cbe.png">

Has something changed? Is it me?

**Comment by mejackreed** [comment]
Hi @rogerhyam 

We made some changes to how the default umd package is bundled and moved the state methods to a different file `mirador-state.min.js` in the same package.

To get you going immediately you could do something like this:

```diff
- store.dispatch(actions.fetchManifest(id));
+ store.dispatch({ manifestId: id, type: 'mirador/FETCH_MANIFEST'});
- store.dispatch(actions.removeManifest(id));
+ store.dispatch({ manifestId: id, type: 'mirador/REMOVE_MANIFEST'});
```

For the long term, I think we will take this under consideration about whether or not to put the state actions into the umd build. Are you using the umd build in your application?

**Comment by rogerhyam** [comment]
I've changed those calls to your suggestion but used 'mirador/REQUEST_MANIFEST' rather than 'mirador/FETCH_MANIFEST' and it loads the manifest but then blows up with "Could not find the drag and drop manager in the context ..." when I try and open it.

I've been adding Mirador to my project with the a simple "npm install mirador" which I'm presuming gets me umd package (not that I knew there was a difference).

I think I need to step back and look at how I'm doing this integration. What is the recommended approach to embedding Mirador 3? I need to be able to add/remove manifests and ideally windows programmatically and at some point in the future receive events from plugins.

Thanks.

**Comment by mejackreed** [comment]
Hi @rogerhyam , the actions have been added back in v3.0.0-rc.4.

> What is the recommended approach to embedding Mirador 3? I need to be able to add/remove manifests and ideally windows programmatically and at some point in the future receive events from plugins.

I think what you are doing is fine. `npm install mirador@rc`. And then using the umd build package at `dist/mirador.min.js`. However, if you find yourself needing to do more complex things like reuse a Mirador 3 component or writing a custom plugin, you may want to consider using the es or cjs modules that are also available in `dist`.

**Comment by HeineDeelstra** [comment]
If I use mirador@rc which currently installs rc5, then import mirador from 'mirador'; in a create react app and obtain an instance from mirador.viewer(config, plugins);, I still do not see an actions property.

Should one use the actions on mirador.default.actions with rc4/5? Those do appear to work.

**Comment by mejackreed** [comment]
Yes the actions on `mirador.default.actions` should be the approach forward. We need to update the documentation to reflect this.
